launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -73,6 +74,12 @@
   /** Logger name to use when launching a child process. */
   public static final String CHILD_PROCESS_LOGGER_NAME = "spark.launcher.childProcLoggerName";
 
+  public static final String LAUNCHER_INTERNAL_PORT = "spark.launcher.internal.port";
```

Comment:
I'm not a fan of having these public variables. This is a public class so we should try to hide anything that is internal to Spark from the API.

A way of doing this would be to leave this as package-private, and move the code that connects to the launcher from `Client.scala` to `YarnCommandBuilderUtils.scala`, since the latter is in the launcher package. And perhaps rename the latter to `YarnLauncherUtils.scala` since it would be doing more than just what it does now.